gateway/admin.go

```diff
@@ -21,3 +31,148 @@ func (g *GatewayAdmin) Status() interface{} {
 	}
 
 }
+
+type GatewayRPCHandler struct {
+	jobsDB jobsdb.JobsDB
+}
+
+type SqlRunner struct {
+	dbHandle     *sql.DB
+	jobTableName string
+	err          error
+}
+
+type SourceEvents struct {
+	Count  int
+	Source string
+}
+
+func (r *SqlRunner) getUniqueSources() []SourceEvents {
+	sources := make([]SourceEvents, 0)
+	if r.err != nil {
```

Comment:
I don't think this is the right way of handling errors. Why can't we use the same standard across everything?

Reply:
Please move this up to the caller.

Reply:
done